packages/adapter-sveltekit/src/lib/checkIsSvelte5.ts

```diff
@@ -0,0 +1,49 @@
+import { SliceMachineContext } from "@slicemachine/plugin-kit";
+import { readProjectFile } from "@slicemachine/plugin-kit/fs";
+import semver from "semver";
+
+import { PluginOptions } from "../types";
+
+export const checkIsSvelte5 = (args: CheckIsSvelteNArgs): Promise<boolean> => {
+	return checkIsSvelteN(5, args);
+};
+
+type CheckIsSvelteNArgs = {
+	helpers: SliceMachineContext<PluginOptions>["helpers"];
+};
+
+const checkIsSvelteN = async (
+	version: number,
+	args: CheckIsSvelteNArgs,
+): Promise<boolean> => {
+	try {
+		const packageJsonContent = await readProjectFile({
+			filename: "package.json",
+			encoding: "utf8",
+			helpers: args.helpers,
+		});
```

Comment:
It might be safer and more reliable to use `await import("svelte/package.json")` or something similar (like `require.resolve`) to read the installed package's version. Using that prevents false negatives when `latest`, `next`, etc. are used.

We used a similar technique elsewhere, but I can't find it. I think it was when we needed to detect when React 17 or 18 was used.